R/clean_data_helpers.R

```diff
@@ -140,66 +140,24 @@ scan_in_character <- function(x, x_name) {
     )
   )
 
-  # when date values are identified, check if they are at the same time numeric
-  # and get the count of ambiguous
-  # convert the rest to numeric and check if they can also translate to data and
-  # get the second count of ambiguous
-  date_count <- ambiguous_count <- numeric_count <- 0L
-  if (sum(!is.na(are_date)) > 0L) {
-    # get the date count and the indices of the date values
-    date_count <- sum(!is.na(are_date))
-    are_date_idx <- which(!is.na(are_date))
+  date_count <- sum(!is.na(are_date))
+  character_count <- character_count - date_count
 
-    # convert the date values into numeric and check if some of them are also
-    # numeric. If some are, get the first count of ambiguous and numeric values
-    are_numeric_in_dates <- suppressWarnings(as.numeric(x[are_date_idx]))
-    ambiguous_count <- sum(!is.na(are_numeric_in_dates))
-    numeric_count <- ambiguous_count
-
-    # getting out of this condition with non-date values
-    character_count <- character_count - date_count
-    x <- x[-are_date_idx]
-
-    # convert the remaining values into numeric.
-    # then check if any of them can be a date value
-    are_numeric <- suppressWarnings(as.numeric(x))
-    numeric_count <- numeric_count + sum(!is.na(are_numeric))
-    are_numeric_idx <- which(!is.na(are_numeric))
-    if (length(are_numeric_idx) > 0L) {
-      numeric_values <- as.numeric(x[are_numeric_idx])
-      x <- x[-are_numeric_idx]
-      character_count <- character_count - length(are_numeric_idx)
-
-      # convert the numeric values into date.
-      # If some are date, get the second count of ambiguous and date values
-
-      # Set the first date to 50 years before the current date
-      oldest_date <- Sys.Date() - lubridate::years(50)
+  # convert everything to numeric and get the numeric count
+  are_numeric <- suppressWarnings(as.numeric(x))
+  numeric_count <- sum(!is.na(are_numeric))
 
-      # identify potential date values and increment date and ambiguous counts
-      date_values <- lubridate::as_date(numeric_values)
-      valid_dates <- date_values >= oldest_date & date_values <= Sys.Date()
```

Comment:
I notice I lost this part but it could be reintroduced in the new logic, without the new for the `if` branches.